GameOfRunes/GameOfRunes/GameEngine/GameEngine.swift

```diff
@@ -145,61 +155,111 @@ class GameEngine {
         systemDelegate.minusHealthPoints(for: entity)
     }
     
-    func enemyReachedLine(_ entity: GKEntity) {
+    func removeEnemy(_ entity: GKEntity) {
         guard let enemyEntity = entity as? EnemyEntity else {
             return
         }
-
         removeDelegate.removeEnemyReachedLine(enemyEntity)
     }
     
+    func enemyReachedLine(_ entity: GKEntity) {
```

Comment:
don't do this, because the logic for removal should be handled by the `removeDelegate`